### DISCUSSION TITLE
What is wrong with my function?

### DISCUSSION THREAD

**Discussion by peppermintbunny** [supabase/supabase#12816] [Questions] [answered]
My function does not work if I include the therapist_id value :  
This does not work!!
---------------------
begin
  insert into public.profiles(
    id,
    email,
    display_name,
    therapist_id,
    kind,
  therapist_display_name,
  raw_user_meta_data)
  values(new.id,
  new.email,
  new.raw_user_meta_data->>'display_name',
  new.raw_user_meta_data->>'therapist_id',
  new.raw_user_meta_data->>'kind',
  new.raw_user_meta_data->>'therapist_display_name',
  new.raw_user_meta_data
  );
  return new;
end;

But the following does work:
----------------------------
begin
  insert into public.profiles(
    id,
    email,
    display_name,
    kind,
  therapist_display_name,
  raw_user_meta_data)
  values(new.id,
  new.email,
  new.raw_user_meta_data->>'display_name',
  new.raw_user_meta_data->>'kind',
  new.raw_user_meta_data->>'therapist_display_name',
  new.raw_user_meta_data
  );
  return new;
end;

Can anyone tell me what is wrong with the first version which  does not work?

**Comment by apikee** [comment]
Is the `therapist_id` value (not property, but value) defined when invoking the function? Meaning there is a value, e.g. "12345"?

**Reply by peppermintbunny** [reply]
Yes, there is a value, its a uuid

**Reply by apikee** [reply]
How does the schema for public.profiles look? Try to check if there is a mismatch in types, `therapist_id` itself should be UUID (in its respective table) so try to check whats the type you specified for `therapist_id` in the public.profiles table

**Comment by GaryAustin1** [comment, marked as answer]
If therapist_id column is a number type column then you need to type the json text when you store it.
`(new.raw_user_meta_data->>'therapist_id')::int8,`  Where int8 is your type of column.

Edit:  Also you can look in the Postgres logs in the dashboard right after your error (assuming you were getting an error) for more info on the issue.

**Reply by peppermintbunny** [reply]
Thank you I'll try that.  :: is new to me.

**Reply by GaryAustin1** [reply]
If uuid then `::uuid`
 would need to be added as ->>'name' is  a text value.

**Reply by peppermintbunny** [reply]
Many thanks.  It works!
